elytron/WFCORE-5278-token-based-authentication.adoc

```diff
@@ -0,0 +1,188 @@
+= [WFCORE-5278] Dynamically issuing a JWT Token during the Elytron authentication process
+:author:            Sonia Zaldana Calles
+:email:             szaldana@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+JSON Web Token (JWT), as defined by RFC7519, is a compact, URL-safe means
+of representing claims to be transferred between two parties. Security tokens in a
+JWT format offer a lightweight and interoperable way to propagate identities
+across different services.
+
+This task involves two main objectives: adding support to
+dynamically issue JWT tokens and generate JWT tokens as part of the
+authentication process to attach them as a private credential to the established
+security identity.
+
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE-5278[WFCORE-5278]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-14409[WFLY-14409]
+* https://issues.redhat.com/browse/ELY-2083[ELY-2083]
+* https://issues.jboss.org/browse/EAP7-1315[EAP7-1315] (This task is just one piece of the RFE, so it's not a full RFE.
+This work will later be pulled into a full RFE.)
+
+
+=== Dev Contacts
+
+* mailto:szaldana@redhat.com[Sonia Zaldana]
+
+=== QE Contacts
+
+N/A
+
+=== Testing By
+
+N/A
+
+This task is not a full RFE, so QE won't be needed yet.
+
+=== Affected Projects or Components
+
+* Elytron
+
+=== Other Interested Projects
+
+N/A
+
+== Requirements
+
+=== Hard Requirements
+
+==== Token Issuance API
+
+One of the two main objectives of this task is to add support to dynamically issue JWT tokens. Aside from token issuance,
+we should also add support to parse JWT tokens by decrypting signed tokens, verifying expiration date and issued at date,
+and parsing and returning claims sets.
+
+Note that we are not writing the API to issue tokens or parse them, as this functionality is already available in the
+https://github.com/smallrye/smallrye-jwt[SmallRye
```

Comment:
I must be missing something. How does it work for EAP Base, where I thought microprofile libraries should be not present?

Reply:
@SoniaZaldana I think we'd just need a dependency on the SmallRye JWT Build API, i.e., `io.smallrye.smallrye-jwt-build`:

https://github.com/smallrye/smallrye-jwt/blob/main/implementation/jwt-build